packages/mendel-middleware/index.js

```diff
@@ -106,7 +106,17 @@ function MendelMiddleware(opts) {
         var pack = bpack({raw: true, hasExports: true});
         var decodedResults = trees.findTreeForHash(params.bundle, params.hash);
         if (!decodedResults || decodedResults.error) {
-            return notFound(res, decodedResults && decodedResults.error);
+            return notFound(res, decodedResults && decodedResults.error, params);
+        }
+
+        var modules = indexedDeps(decodedResults.deps.filter(Boolean));
+
+        if (!modules.length) {
+            // Something wrong, modules shouldn't be zero
+            return notFound(res, {
+                code: 'EMPTYBUNDLE',
+                message: 'Tree contents are empty'
+            }, params);
```

Comment:
Humm, I don't think this is a good idea to do. This is only hiding the issue not solving it. There should be no way to get zero modules from the deps. If you fall into this condition, it is a bug somewhere that should not happen. This is uncovering something way more serious. There could be two things happening:

1. It could be a distributed problem, you might be getting `/mendel/*` requests on the old version of the app, while the new app is being deployed and the user got the root route `/` (the HTML) from the new box with the new hash. If this is the case, the fix is not here on the middleware.

2. You might have a bad manifest. Did you run the `validate-manifest` file against the manifests you are deploying? The script should work. It is inside `mendel-development` package but no one is requiring it. If the manifest is invalid, it might be causing the resolution to be non-deterministic and therefore create empty bundles only sometimes.

3. Somewhere in the code you might be mutating the manifest. It is kinda easy to know if this is happening, you can use `Object.freeze()` after reading the manifest. You can even do this using your Docker infrastructure and T-traffic production to this box. This way you test production without breaking production.

Either way, I don't think the fix is here. There are error codes for 404 and 500 in `mendel-core`. Maybe some of those are trowing and error is being swallowed?